### PR TITLE
CEECORE-760, App archetype make mocha optional add on

### PR DESCRIPTION
**Pull request by scrajesh**

### CODE REVIEW

File: packages/electrode-archetype-opt-mocha/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Electrode Archetype Option Mocha
+
+Allow [Electrode](https://github.com/electrode-io/electrode) apps to optionally choose to skip installing dependencies for jest.
```
Comment (samsel) [inline packages/electrode-archetype-opt-mocha/README.md]: ~Allow~ Allows

File: packages/electrode-archetype-opt-mocha/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    mocha: false
+  }
+};
+```
+
+If nothing is set, then default is to install.
+
+# Usage
+
+This module generally is included by other Electrode modules. An Electrode app don't need to install this directly.
```
Comment (samsel) [inline packages/electrode-archetype-opt-mocha/README.md]: ~don't~ doesn't

File: packages/electrode-archetype-opt-mocha/index.js [hunk 3]
```diff
@@ -0,0 +1 @@
+// empty
```
Comment (samsel) [inline packages/electrode-archetype-opt-mocha/index.js]: what is the need for an empty file?

File: packages/electrode-archetype-opt-mocha/optional-check.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
+"use strict";
+
+const Path = require("path");
+const assert = require("assert");
+
+const cwd = process.env.PWD || process.cwd();
+
+function lookupAppDirByInitCwd() {
```
Comment (samsel) [inline packages/electrode-archetype-opt-mocha/optional-check.js]: @scrajesh can we pls start DRY-ing this logic? It seems to be present in multiple places

File: packages/electrode-archetype-opt-mocha/optional-check.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  if (userConfig === expectTag) {
+    console.log(`${name}: archetype config set ${optionalTagName} to ${userConfig} - installing`);
+    process.exit(0);
+  } else {
+    console.log(
+      `${name}: archetype config set ${optionalTagName} to ${userConfig} - skipping install because it's not ${expectTag}`
+    );
+  }
+} catch (e) {
+  if (defaultInstall === true) {
```
Comment (samsel) [inline packages/electrode-archetype-opt-mocha/optional-check.js]: do we usually write tests for packages?

File: packages/electrode-archetype-opt-mocha/README.md (hunk 1 above)
Comment (divyakarippath) [inline packages/electrode-archetype-opt-mocha/README.md]: replace jest with mocha

File: packages/electrode-archetype-react-app-dev/package.json [hunk 6]
```diff
@@ -97,7 +97,6 @@
     "mime": "^1.0.0",
     "mini-css-extract-plugin": "^0.7.0",
     "mkdirp": "^0.5.1",
-    "mocha": "^4.0.0",
```
Comment (divyakarippath) [inline packages/electrode-archetype-react-app-dev/package.json]: Need to add same changes for electrode-archetype-react-component-dev also

File: packages/electrode-archetype-opt-mocha/README.md (hunk 1 above)
Comment (scrajesh) [reply packages/electrode-archetype-opt-mocha/README.md]: done

File: packages/electrode-archetype-opt-mocha/README.md (hunk 2 above)
Comment (scrajesh) [reply packages/electrode-archetype-opt-mocha/README.md]: done

File: packages/electrode-archetype-opt-mocha/index.js (hunk 3 above)
Comment (scrajesh) [reply packages/electrode-archetype-opt-mocha/index.js]: was a placeholder and now it exports an empty object.
just to make sure the import works properly to setup mocha.

File: packages/electrode-archetype-opt-mocha/README.md (hunk 1 above)
Comment (scrajesh) [reply packages/electrode-archetype-opt-mocha/README.md]: done

File: packages/electrode-archetype-react-app-dev/package.json (hunk 6 above)
Comment (scrajesh) [reply packages/electrode-archetype-react-app-dev/package.json]: good catch! done.

**Comment by scrajesh** [comment]
> In archetype-react-app/arch-clap.js, make the changes to run "test-server-dev" which runs mocha tests only when this option is enabled. Similar changes are already done for Karma and Jest.

-- have made the relevant changes and verified as well.

File: packages/electrode-archetype-opt-mocha/optional-check.js (hunk 4 above)
Comment (scrajesh) [reply packages/electrode-archetype-opt-mocha/optional-check.js]: can do but in a diff story and sprint.

File: packages/electrode-archetype-opt-mocha/optional-check.js (hunk 5 above)
Comment (scrajesh) [reply packages/electrode-archetype-opt-mocha/optional-check.js]: not for all and not sure for this..

**Pull request merged**